### ISSUE TITLE
Gravatar Upload: Display correct error message (possibly immediately) for images that are too large

### ISSUE THREAD

**Issue by v18** [Automattic/wp-calypso#16743]
Steps to repro:
1. Go to /me
2. Upload [this image](https://cloudup.com/cyWmyC4fRrY) as the new Gravatar
3. See the error "Hmm, your new Gravatar was not saved"
4. In the console, see error "413 Request Entity Too Large"

**Comment by mashmac2** [comment]
#560552-chat also had this issue a 5 MB image, resizing it smaller (1000 px square, 50 Kb) worked.

**Comment by cecilerainon** [comment]
Another report through Zendesk: https://woothemes.zendesk.com/agent/tickets/628644
I'm asking for clarification on the picture they are using.

Note that I get the same error message with a picture that's 18Kb. :/

**Comment by v18** [comment]
> Note that I get the same error message with a picture that's 18Kb

@cecilerainon can you list the steps you used to see this error message, and also share a screenshot of the network tab in the console dev tools for the gravatar-upload network request? Feel free to DM me with any info/questions as well.

**Comment by cecilerainon** [comment]
Sure! Steps on both SU and SSP:

- Go to WordPress.com/me
- Click on upload avatar and choose pic
- Error message

Here is a GIF I made, where I try through SU with that 18Kb picture on the user's account and then try on my own account where it works fine: http://cld.wthms.co/baeVYW

As you can see, there's nothing showing up in the console.

**Comment by v18** [comment]
@cecilerainon thank you for the details & video! I still can't repro. Are you able to reproduce this using:

1. Using another image on the same account?
2. Another account through SU?
3. On your own test account, not through SU?

For this case you're able to reproduce, can you see what error message you see in the network tab (you have to make sure to go to the network tab before starting the upload). It'll look like this:

<img width="1114" alt="gravatar-upload-network" src="https://user-images.githubusercontent.com/11487924/29278340-bcf667d4-8114-11e7-9973-717bd258210d.png">

**Comment by cecilerainon** [comment]
I'm sorry @v18, this got buried in my mailbox and I didn't get to it til then. It looks like the user was able to finally upload a picture.

I did however try n.2 and I was able to reproduce the error I had on that user account.

This is the error I got: 
![capture d ecran 2017-09-04 a 11 13 00](https://user-images.githubusercontent.com/16645645/30019823-1fa3f60a-9162-11e7-8a9a-7d4cb08c2a76.png)

**Comment by v18** [comment]
@cecilerainon thanks for trying the steps. I think the issue you're seeing in SU might be unrelated to the Gravatar upload, and be a result of how user data is handled in SU. It should be opened as a separate issue on GH, and I'll table looking into it now.

**Comment by aisajib** [comment]
A user at 1420601-zen reported the same issue. They were seeing the error, "Hmm, your 
new Gravatar was not saved. Please try uploading again." I had them send the image (it's attached on the ZD ticket) and tried it myself. I saw the error too. After I sent them a resized version that's below 1 mb, it worked without an issue. Noting here for future reference.

**Comment by htdat** [comment]
The maximum size of images is 1MB. 

See more in 2813-gh-jpop-issues

**Comment by arunsathiya** [comment]
Confirming that this is still an issue. I tested it a minute ago and files do not appear from https://wordpress.com/me but the same file can be set as display photo on gravatar.com.

**Comment by arunsathiya** [comment]
#8444847-zen

**Comment by tbradsha** [comment]
Reported again here: 13132665-hc

**Comment by jp-imagines** [comment]
Another report here: 3955910-zd

Confirmed via browser console's network tab the same "Invalid account address" error as above. User's email on WP.com and in ZD don't exactly match; could that be related?

**Comment by Sheikhlipu123** [comment]
same problem https://www.awesomescreenshot.com/image/13890270?key=d63946fe4457eb2674291b4b0259dbf5

**Comment by tbradsha** [comment]
Still an issue.
